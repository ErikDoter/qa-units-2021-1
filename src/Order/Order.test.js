/* eslint-disable import/first */
jest.mock('../utils/getDate');
import React from 'react'
import {getDate} from "../utils/getDate";
import {shallow, configure} from 'enzyme';
import Order from "./Order";
import {fakeOrders} from '../data/fakeOrders';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Order.js', () => {

    getDate.mockReturnValue(`22 декабря 2020 года`);

    beforeEach(() => {
        jest.resetModules();
    });

    afterAll(() => {
        jest.resetModules();
    });

    it('Right', () => {
        const wrapper = shallow(<Order order={fakeOrders[1]}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('order is null', () => {

        const order = {
            shop: null,
            date: null
        }
        const wrapper = shallow(<Order order={order}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('empty order', () => {
        const emptyWrapper = shallow(<Order/>);
        expect(emptyWrapper).toMatchSnapshot();
    });
})
