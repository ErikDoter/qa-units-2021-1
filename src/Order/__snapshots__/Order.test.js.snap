// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js Right 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Lamodник.ru
    </span>
    <span
      id="date"
    >
      22 декабря 2020 года
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Жакет - BOREAL5
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Miss Gabby Костюм
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Ostin перчатки мужские
    </div>
    <div
      className="Order-item"
      key="3"
    >
      Zara худи роз.
    </div>
  </div>
</div>
`;

exports[`Order.js empty order 1`] = `""`;

exports[`Order.js order is null 1`] = `""`;
